datafusion/physical-plan/src/joins/symmetric_hash_join.rs

```diff
@@ -337,10 +337,6 @@ impl ExecutionPlan for SymmetricHashJoinExec {
         Ok(children.iter().any(|u| *u))
     }
 
-    fn benefits_from_input_partitioning(&self) -> Vec<bool> {
-        vec![false, false]
-    }
```

Comment:
I don't know why this sets all false for `SymmetricHashJoinExec`. When the mode is `StreamJoinPartitionMode::Partitioned`, it requires input distributions to be `HashPartitioned`s. Based on the current logic, we don't need to override default function.

Current test case also proves the behavior is expected.

Reply:
Roundbound and hashpartition together remove the input order of the SHJ. This is why we added this guard, but it was long ago. I will send a fixing PR for this purpose. If these changes do not affect your tests, can you please keep it as it is and let me delete it?

Reply:
> Roundbound and hashpartition together remove the input order of the SHJ. 

I don't see it requires input ordering (i.e., `required_input_ordering`).

Reply:
Okay. Sounds good.